server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/MockDiskUsagesIT.java

```diff
@@ -215,6 +221,87 @@ public void testAutomaticReleaseOfIndexBlock() throws Exception {
         assertSearchHits(client().prepareSearch("test").get(), "1", "3");
     }
 
+    public void testCancelExistingRecoveriesOnDiskPassingHighWatermark() throws Exception {
```

Comment:
This test only checks the `INDEX_CREATED` case. I suggested combining the two cases, in which case this test is sufficient, but if we don't do that then we need another test for the other case.